### ISSUE TITLE
regression: docker-compose environment variables being altered

### ISSUE THREAD

**Issue by bobzoller** [convox/rack#1225]
#1044 introduced environment variable interpolation for docker-compose.yml, but it has caused a regression in my production apps that contain `$` characters in their `environment` variables.

before, this docker-compose.yml worked:

```yaml
web:
  image: foo
  command: bar
  environment:
    - SEKRET=foo$bar
```

but after #1044 this is sent to CloudFormation as `SEKRET=foo`.

I'd argue applying interpolation to the raw YAML string as a whole is too naive a solution, and it should instead be applied to specific properties post-parsing.  Else, it'd be awesome to make a bunch of noise about such upgrade issues -- I'm now going through and double-quoting all of our environment variables...

**Comment by bobzoller** [comment]
I think there's a second issue -- a double interpolation bug.

in [cmd/build/main.go line 40](https://github.com/convox/rack/blob/master/cmd/build/main.go#L60) we `manifest.LoadFile`, which returns the interpolated manifest, and then on [line 63](https://github.com/convox/rack/blob/master/cmd/build/main.go#L63) we `manifest.Raw` and write it to Dynamo.  From here on any code that uses `manifest.Load` on the `release.Manifest` is double interpolating.

so a yaml like:
```yaml
web:
  image: foo
  command: bar
  environment:
    - SEKRET=foo$$bar
```

ends up in Dynamo as `SEKRET=foo$bar`

which ends up in CloudFormation as `SEKRET=foo`

I verified this by putting `SEKRET=foo$$$$bar` in my docker-compose.yml and verifying that my running app now yields `SEKRET=foo$bar`.

**Comment by mattmanning** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;bzbGUjN2&#x2F;713-docker-compose-yml-env-interpolation-issues"><img src="https:&#x2F;&#x2F;github.trello.services&#x2F;images&#x2F;trello-icon.png" width="12" height="12"> docker-compose.yml env interpolation issues</a>

**Comment by ddollar** [comment]
The double-interpolation is fixed in #1231. Will look at possibly reducing the scope of the substitution as well.
